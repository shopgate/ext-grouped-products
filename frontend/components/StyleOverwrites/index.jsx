import React from 'react';

import sheetStyles from '@shopgate/pwa-ui-shared/Sheet/style';
import { themeConfig } from '@shopgate/pwa-common/helpers/config';

const { variables } = themeConfig;

/**
 * The StyleOverwrites components is supposed to
 * extend and overwrite theme component classes.
 * @return {JSX}
 */
const StyleOverwrites = () => (
  <style>
    {`
.${sheetStyles.container} { border-radius: 10px 10px 0 0; }
.${sheetStyles.content} { max-height: calc(100vh - ${variables.navigator.height + 15}px - var(--safe-area-inset-top)); }
    `}
  </style>
);

export default StyleOverwrites;
