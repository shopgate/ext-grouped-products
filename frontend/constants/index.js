export const SHOPGATE_CATALOG_GET_PRODUCT_CHILDREN = 'shopgate.catalog.getProductChildren';

export const SHOW_ADD_TO_CART_BAR = 'SHOW_ADD_TO_CART_BAR';
export const HIDE_ADD_TO_CART_BAR = 'HIDE_ADD_TO_CART_BAR';

export const REQUEST_PRODUCT_CHILDREN = 'REQUEST_PRODUCT_CHILDREN';
export const RECEIVED_PRODUCT_CHILDREN = 'RECEIVED_PRODUCT_CHILDREN';
export const ERROR_NO_PRODUCT_CHILDREN = 'ERROR_NO_PRODUCT_CHILDREN';

export const ADD_TO_CART_BUTTON_TYPE_DEFAULT = 'default';
export const ADD_TO_CART_BUTTON_TYPE_GROUPED = 'grouped';

export const THEME_GMD = 'gmd';
export const THEME_IOS = 'ios';
