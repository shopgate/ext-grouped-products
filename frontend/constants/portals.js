// Portals
const BEFORE = 'before';
const AFTER = 'after';

export const GROUPED_PRODUCT = 'grouped-product';
export const GROUPED_PRODUCT_BEFORE = `${GROUPED_PRODUCT}.${BEFORE}`;
export const GROUPED_PRODUCT_AFTER = `${GROUPED_PRODUCT}.${AFTER}`;

export const GROUPED_PRODUCT_IMAGE = `${GROUPED_PRODUCT}.image`;
export const GROUPED_PRODUCT_IMAGE_BEFORE = `${GROUPED_PRODUCT}.image.${BEFORE}`;
export const GROUPED_PRODUCT_IMAGE_AFTER = `${GROUPED_PRODUCT}.image.${AFTER}`;

export const GROUPED_PRODUCT_META = `${GROUPED_PRODUCT}.meta`;
export const GROUPED_PRODUCT_META_BEFORE = `${GROUPED_PRODUCT}.meta.${BEFORE}`;
export const GROUPED_PRODUCT_META_AFTER = `${GROUPED_PRODUCT}.meta.${AFTER}`;

export const GROUPED_PRODUCT_PRICE = `${GROUPED_PRODUCT}.price`;
export const GROUPED_PRODUCT_PRICE_BEFORE = `${GROUPED_PRODUCT}.price.${BEFORE}`;
export const GROUPED_PRODUCT_PRICE_AFTER = `${GROUPED_PRODUCT}.price.${AFTER}`;

export const GROUPED_PRODUCT_NAME = `${GROUPED_PRODUCT}.name`;
export const GROUPED_PRODUCT_NAME_BEFORE = `${GROUPED_PRODUCT}.name.${BEFORE}`;
export const GROUPED_PRODUCT_NAME_AFTER = `${GROUPED_PRODUCT}.name.${AFTER}`;

export const GROUPED_PRODUCT_AVAILABILITY = `${GROUPED_PRODUCT}.availability`;
export const GROUPED_PRODUCT_AVAILABILITY_BEFORE = `${GROUPED_PRODUCT}.availability.${BEFORE}`;
export const GROUPED_PRODUCT_AVAILABILITY_AFTER = `${GROUPED_PRODUCT}.availability.${AFTER}`;

export const GROUPED_PRODUCT_CTA = `${GROUPED_PRODUCT}.cta`;
export const GROUPED_PRODUCT_CTA_BEFORE = `${GROUPED_PRODUCT}.cta.${BEFORE}`;
export const GROUPED_PRODUCT_CTA_AFTER = `${GROUPED_PRODUCT}.cta.${AFTER}`;

export const GROUPED_PRODUCT_ADD_TO_CART = `${GROUPED_PRODUCT}.add-to-cart`;
export const GROUPED_PRODUCT_ADD_TO_CART_BEFORE = `${GROUPED_PRODUCT}.add.${BEFORE}`;
export const GROUPED_PRODUCT_ADD_TO_CART_AFTER = `${GROUPED_PRODUCT}.add.${AFTER}`;

export const GROUPED_PRODUCT_FAVORITES = `${GROUPED_PRODUCT}.favorites`;
export const GROUPED_PRODUCT_FAVORITES_BEFORE = `${GROUPED_PRODUCT}.favorites.${BEFORE}`;
export const GROUPED_PRODUCT_FAVORITES_AFTER = `${GROUPED_PRODUCT}.favorites.${AFTER}`;
