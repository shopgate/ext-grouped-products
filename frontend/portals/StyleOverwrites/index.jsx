import React from 'react';

import sheetStyles from '@shopgate/pwa-ui-shared/Sheet/style';
import { themeConfig } from '@shopgate/pwa-common/helpers/config';
import { isGmdTheme } from '../../helpers/index';

const { variables } = themeConfig;

/**
 * The StyleOverwrites components is supposed to
 * extend and overwrite iOS theme css classes.
 * @return {JSX}
 */
const StyleOverwrites = () => {
  if (isGmdTheme()) {
    return null;
  }

  return (
    <style>
      {`
  .${sheetStyles.container} { border-radius: 10px 10px 0 0; }
  .${sheetStyles.content} { max-height: calc(100vh - ${variables.navigator.height + 15}px - var(--safe-area-inset-top)); }
      `}
    </style>
  );
};

export default StyleOverwrites;
