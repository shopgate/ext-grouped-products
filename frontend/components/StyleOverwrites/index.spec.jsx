import React from 'react';
import { shallow } from 'enzyme';
import { THEME_GMD, THEME_IOS } from '../../constants';
import StyleOverwrites from './index';

describe('<StyleOverwrites />', () => {
  it('should render as expected when the gmd theme is not active', () => {
    global.process.env.THEME = THEME_IOS;
    const component = shallow(<StyleOverwrites />);
    expect(component).toMatchSnapshot();
    expect(component.find('style').exists()).toBe(true);
  });

  it('should not render when the gmd theme is active', () => {
    global.process.env.THEME = THEME_GMD;
    const component = shallow(<StyleOverwrites />);
    expect(component).toMatchSnapshot();
    expect(component.find('style').exists()).toBe(false);
  });
});
