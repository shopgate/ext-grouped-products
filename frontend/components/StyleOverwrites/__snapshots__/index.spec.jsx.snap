// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StyleOverwrites /> should not render when the gmd theme is active 1`] = `""`;

exports[`<StyleOverwrites /> should render as expected when the gmd theme is not active 1`] = `
<style>
  
  .css-h69706 { border-radius: 10px 10px 0 0; }
  .css-1aon90l { max-height: calc(100vh - 71px - var(--safe-area-inset-top)); }
      
</style>
`;
